Orange/base.py

```diff
@@ -22,14 +62,15 @@ class Learner:
     learner_adequacy_err_msg = ''
 
     def __init__(self, preprocessors=None):
-        if preprocessors is None:
-            preprocessors = type(self).preprocessors
-        self.preprocessors = list(preprocessors)
+        self.use_default_preprocessors = False
+        if isinstance(preprocessors, Iterable):
+            self.preprocessors = preprocessors
```

Comment:
Yes, but the processors passed in the constructor should not be generators. Also, `self.preprocessors` isn't a generator, it's a tuple. Only `self.active_preprocessors` is a generator. Unless we want to support generators being passed into the constructor?